Directory.Build.props

```diff
@@ -0,0 +1,18 @@
+<Project>
+  <PropertyGroup>
+
+    <LangVersion>7.2</LangVersion>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <IsPackable>false</IsPackable>
+    
+  </PropertyGroup>
+  <ItemGroup>
+    <!-- PrivateAssets="All" is on these packages to ensure they're development dependencies
+         and aren't included as transitionary dependencies to package consumers.
+         For clarity, without PrivateAssets marked here, anyone consuming Microsoft.Identity.Client
+         would also be forced to install these dependencies.  PrivateAssets avoids this problem. -->
+    <PackageReference Include="AsyncUsageAnalyzers" Version="1.0.0-alpha003" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3" PrivateAssets="All" />
```

Comment:
If you use the 5.0 .NET SDK, this package is not necessary (nor recommended). By default, analysis is not enabled for .NET 5.0, but you can enable it for older TFMs by using `<EnableNETAnalyzers>true</EnableNETAnalyzers>`.

Here's an example: https://github.com/openiddict/openiddict-core/pull/1281/files